Make it possible to install node-win-shortcut on non Windows OS

I am trying to use node-win-shortcut as part of electron app where I will have a single code base to be used across all platforms. Right now, when I try to install it on non-windows machine, I get following error message.

```
 npm install node-win-shortcut --save

> node-win-shortcut@0.0.2 install /home/hayashis/git/kdinstall/node_modules/node-win-shortcut
> node-gyp rebuild

make: Entering directory '/home/hayashis/git/kdinstall/node_modules/node-win-shortcut/build'
  CXX(target) Release/obj.target/node_win_shortcut_bindings/node_win_shortcut_bindings.o
../node_win_shortcut_bindings.cc:6:21: fatal error: Windows.h: No such file or directory
 #include <Windows.h>
```

Is there anyway to update it so that, node-win-shortcut behaves similar to how node-winreg behaves? node-winreg allows user to install it on non windows machine and basically does nothing (or errors out cleanly?) if invoked on non-Windows machine.

--- a/.gitignore
+++ b/.gitignore
@@ -12,3 +12,5 @@ logs
 results
 
 npm-debug.log
+node_modules/
+build/

--- a/binding.gyp
+++ b/binding.gyp
@@ -3,7 +3,8 @@
     {
       "target_name": "node_win_shortcut_bindings",
       "sources": [ "node_win_shortcut_bindings.cc"],
-      'libraries': [ '-lruntimeobject.lib','-lshlwapi.lib' ]
+      "include_dirs": [ "<!(node -e \"require('nan')\")" ],
+      "libraries": [ "-lruntimeobject.lib","-lshlwapi.lib" ]
     }
   ]
 }

--- a/node_win_shortcut_bindings.cc
+++ b/node_win_shortcut_bindings.cc
@@ -1,7 +1,6 @@
 #define NTDDI_VERSION 0x06010000
 
-#include <node.h>
-#include <v8.h>
+#include "nan.h"
 
 #include <Windows.h>
 #include <sal.h>
@@ -111,38 +110,34 @@ HRESULT TryCreateShortcut(Handle<Value> exePath, Handle<Value> shortcutName, Han
   return hr;
 }
 
-Handle<Value> CreateShortcut(const Arguments& args)
+NAN_METHOD(CreateShortcut)
 {
-  HandleScope scope;
+  Nan::HandleScope scope;
 
-  if (args.Length() < 3 || !args[0]->IsString() || !args[1]->IsString() || !args[2]->IsString())
+  if (info.Length() < 3 || !info[0]->IsString() || !info[1]->IsString() || !info[2]->IsString())
   {
-    ThrowException(Exception::Error(String::New("Bad usage, expected arguments are: path to exe[string], shortcut name[string], app id [string]")));
-    return scope.Close(Undefined());
+    Nan::ThrowTypeError("Bad usage, expected arguments are: path to exe[string], shortcut name[string], app id [string]");
+    return;
   }
 
   HRESULT hr = CoInitializeEx(NULL, COINITBASE_MULTITHREADED);
 
   if (SUCCEEDED(hr))
   {
-    hr = TryCreateShortcut(args[0], args[1], args[2]);
+    hr = TryCreateShortcut(info[0], info[1], info[2]);
     CoUninitialize();
   }
 
   if (!SUCCEEDED(hr))
   {
-    ThrowException(Exception::Error(String::Concat(
-      String::New("Failed to create shortcut, HRESULT:"),
-      Integer::New(hr)->ToString())));
-    return scope.Close(Undefined());
+    Nan::ThrowError(Nan::New(hr)->ToString());
+    return;
   }
-
-  return scope.Close(Undefined());
 }
 
-void init(Handle<Object> exports) {
-  exports->Set(String::NewSymbol("createShortcut"),
-    FunctionTemplate::New(CreateShortcut)->GetFunction());
+void Init(Local<Object> exports) {
+  exports->Set(Nan::New("createShortcut").ToLocalChecked(),
+               Nan::New<FunctionTemplate>(CreateShortcut)->GetFunction());
 }
 
-NODE_MODULE(node_win_shortcut_bindings, init)
+NODE_MODULE(node_win_shortcut_bindings, Init)

--- a/package.json
+++ b/package.json
@@ -3,7 +3,9 @@
   "version": "0.0.2",
   "description": "Create windows shortcuts using node.js",
   "main": "index.js",
-  "dependencies": {},
+  "dependencies": {
+    "nan": "^2.4.0"
+  },
   "devDependencies": {},
   "keywords": [
     "shortcut",
@@ -12,6 +14,8 @@
     "appid"
   ],
   "scripts": {
+    "install": "node-gyp rebuild",
+    "build": "node-gyp build",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "repository": {

--- a/package.json
+++ b/package.json
@@ -13,6 +13,9 @@
     "winrt",
     "appid"
   ],
+  "os": [
+    "win32"
+  ],
   "scripts": {
     "install": "node-gyp rebuild",
     "build": "node-gyp build",

